documentation/modules/exploit/multi/http/atlassian_crowd_pdkinstall_plugin_upload_rce.md

```diff
@@ -0,0 +1,91 @@
+## Vulnerable Application
+
+Atlassian Crowd 2.1.x prior to 3.0.5, 3.1.x prior to 3.1.6, 3.2.x prior to 3.2.8, 3.3.x prior to 3.3.5, and 3.4.x prior to 3.4.4
+had an incorrect installation of the pdkinstall development plugin, which allows unauthenticated remote attackers to upload and install
+arbitrary plugins on vulnerable Atlassian Crowd and Atlassian Crowd Data Center installations via a POST request to the
+`/<crowd install base>/admin/uploadplugin.action` page.
+
+Successful exploitation results in remote code execution as the user running the Atlassian Crowd server.
+
+To set up a vulnerable installation:
+1. Grab a copy of Windows, any version will do but I tested on Windows 10.
+1. Grab a copy of the 3.0.3 version of Atlassian Crowd from https://product-downloads.atlassian.com/software/crowd/downloads/atlassian-crowd-3.0.3.zip
+1. Grab a copy of the Java 8 JDK from https://download.java.net/openjdk/jdk8u41/ri/openjdk-8u41-b04-windows-i586-14_jan_2020.zip
+1. Unzip the JDK to a target directory.
+1. Edit `JAVA_HOME` environment variable and set it to the location where you extracted the Java 8 JDK.
+1. Update the `PATH` environent variable to include a path to the same location as `JAVA_HOME`, but with `\bin` at the end of it.
+1. Extract the Atlassian Crowd zip.
+1. Edit the `atlassian-crowd-3.0.3\apache-tomcat\bin\catalina.bat` file and remove all the `-Djava.endorsed.dirs` options.
+1. Run `start_crowd.bat`
+1. After a few minutes browse to `http://localhost:8095/crowd/` and click the blue button to do a new install.
+1. Follow the default settings, including using the embedded database.
+1. Browse to `http://localhost:8095/crowd/admin/uploadplugin.action`
+1. *Verify* a 400 error message stating that the page requires a `POST` request is shown.
+
+## Verification Steps
+
+1. Follow the steps in `Vulnerable Application`
+1. `use exploit/multi/http/atlassian_crowd_pdkinstall_plugin_upload_rce`
+1. Set `RHOST` to the target IP
+1. Set `LHOST` to your local IP
+1. `exploit`
+1. You should get a shell back as the user running the Atlassian Crowd application.
+
+## Options
+
+## Scenarios
+
+### Atlassian Crowd 3.0.3 on Windows 10 20H2
```

Comment:
Was this tested on Linux? There's an [official Docker image](https://hub.docker.com/r/atlassian/crowd).

Reply:
Hmm no I did try installing it on Linux but ran into some issues, I'll try the Docker image though 👍

Reply:
Would be a good idea, since this is in `multi`, and Linux is a common platform.

Reply:
Agreed, will test it out there and update details.

Reply:
Fixed with https://github.com/rapid7/metasploit-framework/pull/15501/commits/0d7d5ab93f5b9bd8afcefdd1863054164922084b